cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/MockCruiseControl.java

```diff
@@ -64,10 +70,6 @@ public static ClientAndServer getCCServer(int port, int pendingCalls) throws IOE
         LogManager.getLogManager().readConfiguration(new ByteArrayInputStream(loggingConfiguration.getBytes(UTF_8)));
 
         ClientAndServer ccServer = new ClientAndServer(port);
-        setupTestResponse(ccServer);
-        setupCCStateResponse(ccServer);
-        setupCCRebalanceResponse(ccServer);
-        setupCCUserTasksResponse(ccServer, pendingCalls);
```

Comment:
instead of deleting all of them here, my idea was about leaving them but being able to override, if needed, in specific tests. Removing them from here, now we need to call at least one `setupXXX` in each test.
I don't know if my idea is going to work because we mock the `ccServer` in the constructor and then we "override" the mock during a test (so for example, reapplying `.when` and so on). Does it cause any problem with mocking? wdyt?

Reply:
So we could do that with any of the _expectations_ (what MockServer calls `.when()`) calls) that aren't parameterised. However, there is no way to "override" already existing expectations so you would need to `ccServer.reset()` and then apply the new ones. Basically as `.reset()` is a blunt tool you would end up needing to reapply any relevant expectations if `.reset` is called anywhere in your test class. At which point we may as well `.reset()` after every test and apply the expectations you need for each test at the start.